publisher/config_change_request/config_change_request.py

```
@@ -0,0 +1,205 @@
+EnqueueRule, RoutingRule, and QueueInfo classes and uses the strategy design pattern.
+"""
+from google.cloud import pubsub_v1
+import constants
+from config_change_request import config_type_pb2
+
+class Context:
+  """Accepts a strategy in order to parse and publish a template.
+  """
+  def __init__(self, strategy):
+    self._strategy = strategy
+
+  def prepare_for_publish(self, template):
+    """Use the selected strategy to attempt to parse the template.
+
+    Args:
+      template (str): the reporters comment of the configuration template
+
+    Returns:
+      bool: True if the comment was in valid template format. False if not.
+    """
+    return self._strategy.parse_template(template)
+
+  def pub(self):
+    """Publish using whatever strategy is selected.
+    """
+    self._strategy.publish()
+
+class ConfigurationTypes():
+  """The parent class of the types of configurations(EnqueueRule, RoutingRule, QueueInfo)"""
+
+  def __init__(self, reporter, config_change_type):
+    self.config_change_type = config_change_type
+    self.config_change_request = config_type_pb2.ConfigChangeRequest()
+    self.config_change_request.reporter = reporter
+
+  def publish(self):
+    """Publish a message to a Pub/Sub topic.
+    """
+    client = pubsub_v1.PublisherClient()
+    topic_path = client.topic_path(constants.PROJECT_ID, constants.TOPIC_NAME)
+    data = self.config_change_request.SerializeToString()
+    client.publish(topic_path, data=data)
+    print("Message", self.config_change_type, "sent!")
+
+class EnqueueRule(ConfigurationTypes):
+  """Defines how EnqueueRule templates should be parsed"""
+
+  def parse_template(self, template):
+    """Parses the template into a Protocol Buffer object that will
+     be ready to send, given it is valid.
+
+    Args:
+      template (str): the reporters comment of the configuration template
+
+    Returns:
+      bool: True if the comment was in valid template format.
+      False if not in valid template format.
```

It's better to return the completed template and raise errors for invalid template cases. And when you call the function you can try-catch and handle the errors.

---

Use python logger for errors before skipping over invalid templates